### DISCUSSION TITLE
Set and state throw type errors

### DISCUSSION THREAD

**Discussion by alexander-densley** [pmndrs/zustand#2319] [Bug report] [answered]
I copied and pasted the code example from the typescript docs on how to set up zustand in a typescript project, but it doesn't work. It throws three type errors on 'by', 'set', and 'state'. See picture for what i mean:

<img width="1129" alt="image" src="https://github.com/pmndrs/zustand/assets/89933800/946be6a8-ceee-40cf-8831-936e925b5def">

**Comment by alexander-densley** [comment]
If I use the combine middleware it get's rid of type errors, but docs say I shouldn't have to use it, so I'm looking for a fix if possible. It seems also that the docs have you import {create} but that throws errors now, now you just import create without curly braces.

**Reply by dbritto-dev** [reply]
Default export is deprecated and now Zustand use named exports so the right way to import is by using the curly brackets

**Reply by alexander-densley** [reply]
Makes sense, any thoughts on the type errors?

**Reply by alexander-densley** [reply]
Wait, then why does doing ancurly brace import throw an error?
<img width="1129" alt="image" src="https://github.com/pmndrs/zustand/assets/89933800/4ac8c49e-0b1f-44e6-a98a-8ced57f2d71d">

**Reply by dbritto-dev** [reply]
@alexander-densley try to use type instead of interface

**Reply by alexander-densley** [reply]
@dbritto-dev didn't work :/
![image](https://github.com/pmndrs/zustand/assets/89933800/00835baa-a737-4fa2-9326-92ebd5dc17f0)

**Reply by alexander-densley** [reply, marked as answer]
Ok I fixed it, and it was pretty dumb. Quit vscode and restarted it, something was just messed up in there somewhere, needed a little refresh I guess
